File: backend/Core/Domain/Entities/AuctionBid.cs
using Core.Domain.IdentityEntities;

namespace Core.Domain.Entities
{
    public class AuctionBid
    {
        public Guid AuctionInfoId { get; set; }
        public Guid UserId { get; set; }
        public decimal BidAmount { get; set; }
        public DateTime BidDate { get; set; }
        public AuctionInfo? AuctionInfo { get; set; }
        public ApplicationUser? User { get; set; }
    }
}

File: frontend/src/types/auctionBid.ts
import { AuctionInfo } from "./auctionInfo";
import { User } from "./user";

export interface AuctionBid {
    auctionInfoId: string;
    userId: string;
    bidAmount: number;
    bidDate: Date;
    auctionInfo: AuctionInfo | null;
    user: User | null;
}

export interface AuctionBidDto {
    auctionInfoId: string;
    userId: string;
    bidAmount: number;
    bidDate: Date;
}